docs/root/intro/arch_overview/http/http_filters.rst

```diff
@@ -148,3 +148,72 @@ the filter config name.
 
 Use of per filter config map is filter specific. See the :ref:`HTTP filter documentation <config_http_filters>`
 for if and how it is utilized for every filter.
+
+.. _arch_overview_http_filters_route_based_filter_chain:
+
+Route based filter chain
+------------------------
+
+There is support for having different filter chains for different routes. There are two different modes for this:
+
+* Disabling parts of filters in the filter chain for specific routes.
+* Setting parts of the filters in the filter chain to be disabled by default and then enabling them for specific
+  routes.
+
+By default, the filter chain is the same for all routes and all filters are enabled. However, they
```

Comment:
FWIW, regardless of whether this PR gets merged or not, the first part of this doc is great, and should be added to describe the use of the `disabled` flag.

Reply:
Will drop the code and only keep the docs change.